addons/mail/models/mail_thread.py

```diff
@@ -4426,16 +4426,20 @@ def _extract_partner_ids_for_notifications(self, message, msg_vals, recipients_d
 
     def _truncate_payload(self, payload):
         """
-        Check the payload limit of 4096 bytes to avoid 413 error return code.
+        Check the payload limit of 3990 bytes to avoid 413 error return code.
         If the payload is too big, we trunc the body value.
+        This limit has been chosen based on the extra bytes added by the encryption
+        using AES128GCM. Taking into account what are the extra bytes added at the header:
+        salt(16) + record_size(4) + key_len(1) + public_key (sender_public_key.len)
         :param dict payload: Current payload to trunc
         :return: The truncate payload;
         """
         payload_length = len(str(payload).encode())
         body = payload['options']['body']
         body_length = len(body)
-        if payload_length > 4096:
-            body_max_length = 4096 - payload_length - body_length
+
+        if payload_length > 3990:
+            body_max_length = max(0, 3990 - payload_length + body_length)
```

Comment:
Aren't we still using the decoded length for the body here?
Can you craft a test that fails like this and passes when using a body that's just a little bit too large?

Or am I wrong and is there no case like that?

Don't be afraid to ping me by the way otherwise I might easily forget about this ^^'
Thanks, cheers